account-kit/infra/src/middleware/gasManager.ts

```diff
@@ -26,3 +49,204 @@ export function alchemyGasManagerMiddleware(
     context: { policyId: policyId },
   });
 }
+
+interface AlchemyGasAndPaymasterAndDataMiddlewareParams {
+  policyId: string;
+  transport: AlchemyTransport;
+  gasEstimatorOverride?: ClientMiddlewareFn;
+  feeEstimatorOverride?: ClientMiddlewareFn;
+}
+
+/**
+ * Paymaster middleware factory that uses Alchemy's Gas Manager for sponsoring transactions using
+ * the `alchemy_requestGasAndPaymasterAndData` method instead of standard ERC-7677 methods.
+ *
+ * @example
+ *  ```ts
+ * import { sepolia, alchemyGasAndPaymasterAndDataMiddleware } from "@account-kit/infra";
+ * import { alchemy } from "@account-kit/infra";
+ *
+ * const client = createAlchemySmartAccountClient({
+ *  transport: alchemy({ apiKey: "your-api-key" }),
+ *  chain: sepolia,
+ *  ...alchemyGasAndPaymasterAndDataMiddleware({
+ *    policyId: "policyId",
+ *    transport: alchemy({ apiKey: "your-api-key" }),
+ *  })
```

Comment:
is this how we intend it to be used as well? Ideally the change to this middleware happens invisibily to users.

Currently, you just pass a `policyId` to `createAlchemySmartAccountClient` and under the hood that makes this call

Reply:
oh just saw the impl above does do that, then I'd update this example to use `createSmartAccountClient` instead of `createAlchemySmartAccountClient`

also worth calling out in the description that someone shouldn't have to use this method directly since the alchemy client will handle this for you

Reply:
oops, great call-out! def should just enable it here by passing the `policyId`.